xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetable.css

```diff
@@ -167,6 +167,7 @@ thead.xwiki-livetable-display-header td.xwiki-livetable-display-header-filter se
   color: $theme.textPrimaryColor;
   margin: 0;
   padding: 0 0 0 2px;
+  margin-top: 2px;
```

Comment:
It's not clear for me why do you need this margin and why don't you apply it only to the bootstrap-select.